src/biome/text/dataset_readers/mixins.py

```diff
@@ -97,17 +100,17 @@ def build_textfield(
             Either a TextField or a ListField containing the record.
             Returns None if `data` is not a str or a dict.
         """
-        if not isinstance(data, (str, list, dict)):
+        if not isinstance(data, (str, Iterable, dict)):
             self._logger.warning(
-                "Cannot process data example %s of type %s", data, type(data)
+                "Processing data %s of type %s as string", data, type(data)
             )
-            return None
+            data = self._value_as_string(data)
```

Comment:
Not sure if this is possible, but if `data` is None, this change would lead to returning a `TextField`/`ListField` with token `None`.